plugins/modules/aci_l3out.py

```diff
@@ -52,14 +57,66 @@
     aliases: [ route_control_enforcement ]
   l3protocol:
     description:
-    - Routing protocol for the L3Out
+    - Routing protocol for the L3Out.
+    - Protocols must be re-provided with their paramaters each time an l3out with existing protocols is modified if the protocols were to be kept.
+    - The Protocols are otherwise deleted if not provided each time an l3out is modified.
+    - First example, to add BGP protocol to an l3out with OSPF protocol, the user must enter C([ bgp, ospf ]) even though "ospf" was provided before.
+    - Second example, to change the protocol from OSPF to EIGRP, the user must simply enter C([ eigrp ]) and the previous OSPF protocol will be deleted.
+    - To remove all existing protocols, the user must enter C([ static ]).
     type: list
     elements: str
     choices: [ bgp, eigrp, ospf, pim, static ]
+  ospf:
+    description:
+    - Parameters for the OSPF protocol.
+    type: dict
+    suboptions:
+      area_cost:
+        description:
+        - The OSPF area cost.
```

Comment:
We typically provide the default setting of apic when variable is not provided, see example https://github.com/CiscoDevNet/ansible-aci/blob/master/plugins/modules/aci_bd.py#L47

```suggestion
        - The OSPF area cost.
        - The APIC defaults to C(1) when unset during creation.
```

Reply:
This applies to all attributes that have a default.